distributed/diagnostics/tests/test_scheduler_plugin.py

```diff
@@ -125,3 +125,16 @@ async def close(self):
 
     assert plugin.history == ["started", "closed"]
     assert plugin.scheduler is s
+
+
+@gen_cluster(client=True)
+async def test_register_scheduler_plugin(c, s, a, b):
+    temp_list = []
+
+    class Dummy(SchedulerPlugin):
+        def start(self, scheduler):
+            temp_list.append(1)
+
+    await c._register_scheduler_plugin(Dummy)
+
+    async_wait_for(temp_list, 1)
```

Comment:
Since `async_wait_for` is a coroutine, we still need to `await` it 

```suggestion
    await async_wait_for(temp_list, 1)
```